### DISCUSSION TITLE
Switch between Tabs

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#13621] [Q&A] [answered]
I have open a TabGroup containing 3 Tabs. Each of them represents his own view and controller (Alloy framework). If I want to switch between the Tabs do I have to call `Alloy.createController('tab2").getView().open();`? Or is there something like `Alloy.Views.tab2.setActive();`?

**Comment by m1ga** [comment, marked as answer]
`tabgroup.activeTab = 1` (https://titaniumsdk.com/api/titanium/ui/tabgroup.html#activetab)

```js
var win1 = Ti.UI.createWindow({
    backgroundColor: 'blue',
    title: 'Blue'
});
win1.add(Ti.UI.createLabel({text: 'click to go to tab 2'}));
win1.addEventListener("click",function(){
	tabGroup.activeTab = 1
})

var win2 = Ti.UI.createWindow({
    backgroundColor: 'red',
    title: 'Red'
});
win2.add(Ti.UI.createLabel({text: 'click to go to tab 1'}));
win2.addEventListener("click",function(){
	tabGroup.activeTab = 0
})
var tab1 = Ti.UI.createTab({
    window: win1,
    title: 'Blue'
}),
tab2 = Ti.UI.createTab({
    window: win2,
    title: 'Red'
}),
tabGroup = Ti.UI.createTabGroup({
    tabs: [tab1, tab2]
});
tabGroup.open();
```

**Reply by beppo-ivel** [reply]
Assume you have a structure like the snippet below, where each Tab represents his own view and controller. For example you have a click event in controller "tab2" and if the event is triggered you want to switch to tab1. How to access the variable `$.tabGroup` in the tab2-controller? Because `$.tabGroup` is only accesible in index-controller.

index.xml:
```
<Alloy>
   <TabGroup id="tabGroup">
      <Require src="tab1" />
      <Require src="tab2" />
      <Require src="tab3" />
   </TabGroup>
</Alloy>
```

**Reply by m1ga** [reply]
in your index do:
```js
Alloy.Globals.tabGroup = $.tabGroup
```

and in your controllers you can use `Alloy.Globals.tabGroup.activeTab = ...`.
Or Backbone events and trigger a change in a controller and listen for it in index

**Reply by beppo-ivel** [reply]
If I would use `Alloy.createController('tab1").getView().open();` then controller and view get fresh initialized?

**Reply by m1ga** [reply]
that would  create a new controller and open that as a window outside of the tabgroup. In a tabgroup you only open the tabgroup
